python/src/scipp/plotting/resampling_model.py

```diff
@@ -120,22 +123,32 @@ def _call_resample(self):
             self._home = self._resampled
             self._home_params = self._resampled_params
 
+    def update_array(self, array):
+        """
+        Update the internal array with a new array.
+        """
+        self._array = array
+
+    def reset_params(self):
+        """
+        Reset view parameters to force a plot update.
+        """
+        self._home_params = None
+        self._resampled_params = None
+
 
 class ResamplingBinnedModel(ResamplingModel):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         # TODO See #1469. This is a temporary hack to work around the
         # conversion of coords to edges in model.py.
-        new_meta = {name: var for name, var in self._array.meta.items()}
-        self._array = PlotArray(data=self._array.data,
-                                masks=self._array.masks,
-                                meta=new_meta)
-        for name, var in self._array.meta.items():
+        self._array = self._array.copy(deep=False)
+        for name, var in self._array.coords.items():
             if len(var.dims) == 0:
                 continue
             dim = var.dims[-1]
-            if name not in self._array.data.bins.meta:
-                self._array.meta[name] = to_bin_centers(var, dim)
+            if name not in self._array.data.bins.coords:
+                self._array.coords[name] = to_bin_centers(var, dim)
```

Comment:
Shouldn't all this be called every time someone uses `update_array()`?

Reply:
It should if the coordinates are also modified.
Right now we are not supporting `redraw` for modified coords.
So I thought it would be cheaper to only do it once.
This might also disappear in the future when we remove the conversion to bin edges.

I could also just add a comment in the code explaining just this.

Reply:
Well, but aren't the shiny modifed coords gone once you call `update_array()`, since that overwrites `self._array`?